graphene_plugin.py

```diff
@@ -180,24 +194,29 @@ def create_attribute_type(attribute_node: AssignmentStmt) -> Optional['TypeNodeI
         return None
 
     argument_node_names = attribute_node.rvalue.arg_names  # type: ignore[attr-defined]
-    return_type_name = get_type_string_from_graphene_type(argument_nodes)
+    return_type = get_python_type_from_graphene_type(ctx, argument_nodes)
 
     argument_list: List['ArgumentNode'] = []
     names_to_nodes = zip(argument_node_names[1:], argument_nodes[1:])
 
     for argument_name, argument_node in names_to_nodes:
         if argument_name in ['description', 'required', 'default_value', 'deprecation_reason']:
             continue
-        type_name = get_type_string_from_graphene_type([argument_node], argument_node_names)
-        argument_list.append(ArgumentNode(name=argument_name, type_name=type_name, context=argument_node))
+        python_type = get_python_type_from_graphene_type(ctx, [argument_node], argument_node_names)
+        argument_list.append(ArgumentNode(name=argument_name, type=python_type, context=argument_node))
 
     return TypeNodeInfo(
         name=name,
-        return_type_name=return_type_name,
+        return_type=return_type,
         arguments=argument_list,
         context=attribute_node,
     )
 
+def are_types_equal(left: Type, right: Type) -> bool:
+    if isinstance(left, AnyType) or isinstance(right, AnyType):
+        return True
+    return left == right
```

Comment:
You said this handles `Union`s right (I think you mentioned it handled an `Optoinal`)? If so it's surprising it doesn't handle `Any` being "equal" to all other types.

Reply:
This handles checking that types are _exactly_ equal to other types. So I think it would actually be surprising if `AnyType` was equal to something that isn't also `AnyType`. This checks that two types are _equal_ not that they're _compatible_.

Reply:
Ah right, makes sense. So `Optional[str] == str` would be false. I think I just misunderstood you before.

Reply:
Oh yeah you were saying that the utility function I wrote about in a comment earlier was the thing that did the covariant style type checking, but `==` gives us actual equality checks. Nvm me.